src/utils.rs

```diff
@@ -1,5 +1,30 @@
 use std::borrow::Cow;
 use std::iter;
+use std::sync::{Arc, LazyLock, Mutex, Weak};
+
+use rustc_hash::FxBuildHasher;
+use weak_table::WeakHashSet;
+
+#[derive(Debug, Default)]
+pub struct SymbolTable(WeakHashSet<Weak<str>, FxBuildHasher>);
+
+impl SymbolTable {
+    pub fn intern<T: Into<Arc<str>> + AsRef<str>>(&mut self, name: T) -> Arc<str> {
+        if let Some(rc) = self.0.get(name.as_ref()) {
+            rc
+        } else {
+            let rc: Arc<str> = name.into();
+            self.0.insert(Arc::clone(&rc));
+            rc
+        }
+    }
+}
```

Comment:
I don't think the type or the method needs to be `pub`.

Reply:
Done 👍